include/Definitions.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file is part of the JKQ DD Package which is released under the MIT license.
+ * See file README.md or go to http://iic.jku.at/eda/research/quantum_dd/ for more information.
+ */
+
+#ifndef DDpackage_DATATYPES_HPP
+#define DDpackage_DATATYPES_HPP
+
+#include <cstdint>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace dd {
+    // integer type used for indexing qubits
+    // needs to be a signed type to encode -1 as the index for the terminal
+    // std::int_fast8_t can at least address 128 qubits as [0, ..., 127]
+    using Qubit = std::int_fast8_t;
```

Comment:
In addition to the documentation, we can add a static_assert.
```
static_assert(std::is_signed_v<Qubit>, "Type Qubit must be signed.");
```
What do you think?

Reply:
Certainly a good idea. I also added such asserts for `RefCount` and `fp`.